src/CommunityToolkit.Maui.Maps/Handler/Map/MapHandler.Windows.cs

```diff
@@ -263,20 +290,36 @@ function removeAllPins()
 								locationPin = null;
 							}
 
-							function addPin(latitude, longitude, label, address)
+							function addPin(latitude, longitude, label, address, id)
 							{
 								var location = new Microsoft.Maps.Location(latitude, longitude);
 								var pin = new Microsoft.Maps.Pushpin(location, {
 								            title: label,
 											subTitle: address
 								        });
+
+								pin.markerId = id;
 								map.entities.push(pin);
 								Microsoft.Maps.Events.addHandler(pin, 'click', function (e) 
 								{
+									if (e.targetType !== 'pushpin')
+										return;
+
+									//Set the infobox options with the metadata of the pushpin.
+									infobox.setOptions({
+										location: e.target.getLocation(),
+										title: e.target.getTitle(),
+										description: e.target.getSubTitle(),
+										visible: true
+									});
+
+									infobox.infoWindowMarkerId = id;
```

Comment:
may be e.target.markerId?

Reply:
but the id is from the outer scope, and its value is unpredicted. If it works, please ignore me and merge the PR!